src/main/kotlin/com/mdrobnak/lalrpop/inspections/WrongInferredTypeInspection.kt

```diff
@@ -1,36 +1,77 @@
 package com.mdrobnak.lalrpop.inspections
 
 import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.LocalInspectionToolSession
 import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
 import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.PsiFileFactory
 import com.mdrobnak.lalrpop.psi.LpNonterminal
 import com.mdrobnak.lalrpop.psi.LpVisitor
+import com.mdrobnak.lalrpop.psi.ext.importCode
+import org.rust.lang.RsLanguage
+import org.rust.lang.core.psi.RsTypeAlias
+import org.rust.lang.core.psi.ext.childrenOfType
+import org.rust.lang.core.resolve.ImplLookup
+import org.rust.lang.core.resolve.knownItems
+import org.rust.lang.core.types.infer.RsInferenceContext
 
 /**
  * Reports 2 kinds of problems on a nonterminal
  * 1. The inferred type is different from the type declared explicitly on the nonterminal
  * 2. There are 2 different alternatives, without action code, where the inferred types are different
  */
 object WrongInferredTypeInspection : LocalInspectionTool() {
-    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+    override fun buildVisitor(
+        holder: ProblemsHolder,
+        isOnTheFly: Boolean,
+        session: LocalInspectionToolSession
+    ): PsiElementVisitor {
         return object : LpVisitor() {
             override fun visitNonterminal(nonterminal: LpNonterminal) {
+                if (nonterminal.nonterminalName.nonterminalParams != null) return
```

Comment:
I think it may be possible to support nonterminals with params by generating unit structs in Rust file. What do you think?

Reply:
I'm not really sure what you mean. `ctx.canCombineTypes()` would return `true` only if both types are `()`, right? What do you have in mind?

PS: sorry for not answering sooner, I was way too busy this week.

Reply:
No worries about responding slower; it's the holiday season now :slightly_smiling_face:.

For each type/rule parameter, the generated Rust code could have a `RuleN` struct which is used in place of the generic type. For example:
```
MyNonterminal<K, V>: HashMap<K, V> = ...; // alternative type resolves to std::collections::HashMap<K, V>
```
```rust
// imports

struct Rule1;
struct Rule2;

type T1 = HashMap<Rule1, Rule2>;
type T2 = std::collections::HashMap<Rule1, Rule2>;
```

There might be some issues if the generic type has bounds, like `Hash` in this case. It may be better to check if the Rust plugin supports comparing types with unbound generics (i.e. `type T1<K, V> = HashMap<K, V>;`).

But I think this can wait for a future PR, if you think it's worthwhile to pursue.